lib/warnless.c

```diff
@@ -227,13 +169,17 @@ unsigned int curlx_uztoui(size_t uznum)
 # pragma warning(push)
 # pragma warning(disable:810) /* conversion may lose significant bits */
 #endif
-
-#if (SIZEOF_INT < SIZEOF_SIZE_T)
-  DEBUGASSERT(uznum <= (size_t) CURL_MASK_UINT);
+#ifdef _MSC_VER
+# pragma warning(push)
+# pragma warning(disable:4127) /* conditional expression is constant */
 #endif
+
+  if(sizeof(int) < sizeof(size_t))
```

Comment:
This could be made into a preprocessor directive;
~~~c
#if UINT_MAX < SIZE_T_MAX

Reply:
> This could be made into a preprocessor directive;
> 
> ```c
> #if UINT_MAX < SIZE_T_MAX
> ```

The reason why I decided using `if(sizeof(int) < sizeof(size_t))` instead of preprocessor defines is to have maximum portability (any complier has 'sizeof`, negate and shift operators), if this or that define is missing, like for example `SIZE_T_MAX` which can be missing in some older MSVC versions, if I am not mistaken, then proposed simplification will run into additional checks for defines existence.

`if(sizeof(int) < sizeof(size_t))` is removed by the compiler with any optimization level and therefore has no effect on performance while being effective check in the same time against types inconsistency.

I can of course replace to compile time preprocessor directives but anxious about portability. What do you think about my reasoning?

Reply:
Thing is, we use those defines in existing curl source code so we already need to have them available, using them in this file is thus not any added portability issue. I realize a compiler can remove the fixed value condition, but I'm concerned about warnings and that the assert is also a null-operation in non-debug builds so it makes a fixed condition to a blank expression.

Reply:
Ok, I see. I will adjust these checks to the preprocessor directives as proposed then.